plugins/lookup/get_moid.py

```diff
@@ -0,0 +1,579 @@
+                "hosts": "hosts",
+                "names": "names",
+                "resource_pools": "resource_pools",
+                "vms": "vms"
+            }
+        }
+    },
+    "network": {
+        "list": {
+            "query": {
+                "datacenters": "datacenters",
+                "folders": "folders",
+                "names": "names",
+                "networks": "networks",
+                "types": "types"
+            }
+        }
+    }
+}
+
+import asyncio
+import os
+import q
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible.plugins.lookup import LookupBase
+
+from ansible_collections.cloud.common.plugins.module_utils.turbo.exceptions import (
+    EmbeddedModuleFailure,
+)
+from ansible_collections.vmware.vmware_rest.plugins.module_utils.vmware_rest import (
+    gen_args,
+    open_session
+)
+
+
+def _get_credentials(**options):
+    credentials = {}
+    credentials['vcenter_hostname'] = options.get('vcenter_hostname') or os.getenv('VMWARE_HOST')
+    credentials['vcenter_username'] = options.get('vcenter_username') or os.getenv('VMWARE_USER')
+    credentials['vcenter_password'] = options.get('vcenter_password') or os.getenv('VMWARE_PASSWORD')
+    credentials['vcenter_validate_certs'] = options.get('vcenter_validate_certs') or os.getenv('VMWARE_VALIDATE_CERTS')
+    credentials['vcenter_rest_log_file'] = options.get('vcenter_rest_log_file') or os.getenv('VMWARE_REST_LOG_FILE')
+
+    return credentials
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables, **kwargs):
+        self.set_options(var_options=variables, direct=_get_credentials(**kwargs))
+        loop = asyncio.get_event_loop()
+        return loop.run_until_complete(asyncio.gather(self.entry_point(terms)))
+
+    async def fetch(self, url):
+        async with self._options['session'].get(url) as response:
+            result = await response.json()
+            return result
+
+    def build_url(self, object_type, params):
+        try:
+            _in_query_parameters = INVENTORY[object_type]["list"]["query"].keys()
+            if object_type == 'resource_pool':
+                object_type = object_type.replace('_', '-')
+        except KeyError:
+            raise AnsibleError(
+                "object_type must be one of [%s]." % ", ".join(list(INVENTORY.keys()))
+            )
+
+        return (f"https://{self._options['vcenter_hostname']}/api/vcenter/{object_type}") + gen_args(params, _in_query_parameters)
+
+    async def get_moid(self, object):
+        object_type, object_path = object.popitem()
+
+        # Split object_path for transversal
+        _objects_list = list(filter(None, object_path.split('/')))
+        objects_list = _objects_list.copy()
+        result = None
+        object_name = None
+        _filter = {}
+
+        # GET datacenter MoID
+        _filter['names'] = _objects_list.pop(0)  # save datacenter name
```

Comment:
I think you can avoid the `_objects_list.copy()` above and pass slice of list `objects_list[1:]` or use index.
You can also use a tuple to be sure you don't modify the list:

```
>>> a = tuple(filter(None, object_path.split('/')))
>>> object_path = "/a/0/afr/False/se/"
>>> a = tuple(filter(None, object_path.split('/')))
>>> a[1:]
('0', 'afr', 'False', 'se')
>>> a[0]
'a'
>>> a.append("b")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'tuple' object has no attribute 'append'
```

Reply:
Used the tuple.